Submission.java

```diff
@@ -181,23 +258,95 @@ private JSONArray getSprites()
      * Get script count for sprite.
      *
      * @param spriteName 
-     * @return scriptCount 
+     * @return count 
      */
     public int getScriptCountForSprite(String spriteName)
     {
+        return getCountForSprite("scripts", spriteName);
+    }
+    
+    /**
+     * Get variable count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getVariableCountForSprite(String spriteName)
+    {
+        return getCountForSprite("variables", spriteName);
+    }
+    
+    /**
+     * Get list count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getListCountForSprite(String spriteName)
+    {
+        return getCountForSprite("lists", spriteName);
+    }
+    
+    /**
+     * Get script comment count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getScriptCommentCountForSprite(String spriteName)
+    {
+        return getCountForSprite("scriptComments", spriteName);
+    }
+
+    /**
+     * Get sound count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getSoundCountForSprite(String spriteName)
+    {
+        return getCountForSprite("sounds", spriteName);
+    }
+
+    /**
+     * Get costume count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getCostumeCountForSprite(String spriteName)
+    {
+        return getCountForSprite("costumes", spriteName);
+    }
+    
+    /**
+     * Get count for sprite.
+     *
+     * @param item 
+     * @param spriteName 
+     * @return count 
+     */
+    private int getCountForSprite(String item, String spriteName)
+    {
         JSONArray sprites = getSprites();
-        JSONArray scripts = new JSONArray();
+        JSONArray items = new JSONArray();
         for (int i = 0; i < sprites.size(); i++)
```

Comment:
@KaraBeason So, this was code I refactored out of all the countForSprite methods. Instead of repeating this code over and over, I pulled it out so the other functions call this and pass it the name of the sprite to get the count for, and the JSON attribute name of the item to count (costumes, scripts, etc). What do you think would be a more descriptive name for the function/variables? It's kind of a generic function, with the specificity being passed in as parameters, so I guess that's why I named it kind of generic - but if you have a suggestion for something that is more self-documenting, I'm happy to change it!

Also, I was passing the wrong array element to one of the tests, which was causing the null pointer exception above. I just pushed a change - take a look and let me know if that fixes it on your end.

Thanks so much!